flink-python/pyflink/common/tests/test_python_option.py

```diff
@@ -0,0 +1,130 @@
+# limitations under the License.
+################################################################################
+import json
+import os
+import tempfile
+
+from py4j.protocol import Py4JError
+from pyflink.util.utils import to_jarray
+
+from pyflink.common import Configuration
+from pyflink.java_gateway import launch_gateway
+
+from pyflink.common.dependency_manager import DependencyManager
+
+from pyflink.testing.test_case_utils import (PyFlinkTestCase, replace_uuid,
+                                             TestEnv, encode_to_base64,
+                                             create_empty_file)
+
+
+class PythonOptionTests(PyFlinkTestCase):
+
+    def setUp(self):
+        self.j_env = TestEnv()
+        self.config = Configuration()
+        self.dependency_manager = DependencyManager(self.config, self.j_env)
+
+    def test_python_options(self):
+        dm = DependencyManager
+
+        system_env = dict()
+        system_env[dm.PYFLINK_PY_FILES] = "/file1.py\nhdfs://file2.zip\nfile3.egg"
+        system_env[dm.PYFLINK_PY_REQUIREMENTS] = "a.txt\nb_dir"
+        system_env[dm.PYFLINK_PY_EXECUTABLE] = "/usr/local/bin/python"
+        system_env[dm.PYFLINK_PY_ARCHIVES] = "/py3.zip\nvenv\n/py3.zip\n\ndata.zip\ndata"
+
+        self.dependency_manager.load_from_env(system_env)
+
+        configs = self.config.to_dict()
+        python_files = replace_uuid(json.loads(configs[dm.PYTHON_FILES]))
+        python_requirements_file = replace_uuid(configs[dm.PYTHON_REQUIREMENTS_FILE])
+        python_requirements_cache = replace_uuid(configs[dm.PYTHON_REQUIREMENTS_CACHE])
+        python_archives = replace_uuid(json.loads(configs[dm.PYTHON_ARCHIVES]))
+        python_exec = configs[dm.PYTHON_EXEC]
+        registered_files = replace_uuid(self.j_env.to_dict())
+
+        self.assertEqual(
+            {"python_file_0_{uuid}": "file1.py",
+             "python_file_1_{uuid}": "file2.zip",
+             "python_file_2_{uuid}": "file3.egg"}, python_files)
+        self.assertEqual(
+            {"python_archive_3_{uuid}": "venv",
+             "python_archive_4_{uuid}": "py3.zip",
+             "python_archive_5_{uuid}": "data"}, python_archives)
+        self.assertEqual("python_requirements_file_6_{uuid}", python_requirements_file)
+        self.assertEqual("python_requirements_cache_7_{uuid}", python_requirements_cache)
+        self.assertEqual("/usr/local/bin/python", python_exec)
+        self.assertEqual(
+            {"python_file_0_{uuid}": "/file1.py",
+             "python_file_1_{uuid}": "hdfs://file2.zip",
+             "python_file_2_{uuid}": "file3.egg",
+             "python_archive_3_{uuid}": "/py3.zip",
+             "python_archive_4_{uuid}": "/py3.zip",
+             "python_archive_5_{uuid}": "data.zip",
+             "python_requirements_file_6_{uuid}": "a.txt",
+             "python_requirements_cache_7_{uuid}": "b_dir"}, registered_files)
+
+    def test_python_options_integrated(self):
```

Comment:
What about removing this test as I think we can add Python integration test in the flink-end-to-end-tests in the future? Personally I think this test is a little complicate and it's difficult to maintain.